datafusion/core/src/physical_optimizer/join_selection.rs

```diff
@@ -136,6 +136,9 @@ fn swap_join_type(join_type: JoinType) -> JoinType {
         JoinType::LeftMark => {
             unreachable!("LeftMark join type does not support swapping")
         }
+        JoinType::RightMark => {
+            unreachable!("RightMark join type does not support swapping")
```

Comment:
I think we can support it now, I think this should be relatively easy.

Reply:
Yeah, supporting swap is partly why we are adding RightMark. So would be good to have that fixed in this PR.